code/game/objects/items/devices/suit_cooling.dm

```diff
@@ -28,14 +28,17 @@
 /obj/item/device/suit_cooling_unit/ui_action_click()
 	toggle(usr)
 
-/obj/item/device/suit_cooling_unit/New()
-	START_PROCESSING(SSobj, src)
-	cell = new/obj/item/weapon/cell/high()	//comes not with the crappy default power cell - because this is dedicated EVA equipment
-	cell.loc = src
+/obj/item/device/suit_cooling_unit/Initialize()
+	. = ..()
+	cell = new/obj/item/weapon/cell/high(src)	//comes not with the crappy default power cell - because this is dedicated EVA equipment
+
+/obj/item/device/suit_cooling_unit/Destroy()
+	qdel_null(cell)
+	return ..()
```

Comment:
Ideally should stop processing when deleted.

Reply:
There's actually an attempt to do that in /obj/Destroy(). All my other optimization ones are double-safety, but, missing one won't destroy the universe.